### PR TITLE
Replace DESCRIPTION.rst with Markdown version of the file

### LINKED ISSUE #857: Updates to DESCRIPTION.rst
The `DESCRIPTION.rst` file in the root of our project directory is the file that determines what is displayed on both PyPI and Anaconda Cloud. This file could use some love as a result (both in terms of content and grammar). 

Additionally, if PyPI supports it, it would be better to have this file as Markdown rather than reStructuredText because Anaconda Cloud requires Markdown and right now we have to manually edit that page. However, PyPI should be the priority for deployment. 

Note that this file is loaded by `setup.py` into the `long_description` field of the setup function.

Todo Items:

- [x] Determine if PyPI will render Markdown in addition to reStructuredText
- [x] Convert `DESCRIPTION.rst` to `DESCRIPTION.md` if yes to above 
- [x] Add new YB galleray image to replace the old one (see below)
- [x] Edit the DESCRIPTION file for both content and grammar 

yellowbrick

### LINKED ISSUE #895: Add affiliations to README and Documentation
This issue is to update our documentation and README.md (and DESCRIPTION.rst) with our affiliations by placing an image badge and a link on those sites. 

NumFOCUS has recently sent out a badge that identifies our project as NumFOCUS affiliated:

> We are pleased to send you (see attached) an official badge identifying your project as a NumFOCUS Affiliated Project!
>
> Please feel welcome and encouraged to post this to your website and/or repo. We ask that you create a link to https://numfocus.org should someone try to click on the image.

AffiliatedProject

I have also sent an email to @ojedatony1616 to see if he was interested in doing something similar for DDL.

### PR DESCRIPTION
**Pull request by Kautumn06**
This PR closes #857 by replacing `DESCRIPTION.rst` with a Markdown version of the file. It also closes #895 by adding new logos for our affiliates.

- Created a Markdown version of `DESCRIPTION.rst` and updated `setup.py` to reference the new file
- Added new logos for NumFOCUS and District Data Labs
- Updated the image gallery, text, and the list of visualizers in `DESCRIPTION.md`

In addition, I made several small changes to both our documentation and in our visualizers' docstrings to fix minor misspellings, formatting issues and broken links.

### CODE REVIEW

**Comment by rebeccabilbro** [comment]
Thank you so much for taking this one on @Kautumn06 &mdash; let me know if there's anything I can do to help!

File: README.md [hunk 1]
```diff
@@ -118,6 +118,6 @@ We would be glad if you used Yellowbrick in your scientific publications! If you
 
 ## Affiliations
 
-[![District Data Labs](docs/images/readme/ddl-logo.png)](https://districtdatalabs.com/)
+[![District Data Labs](docs/images/readme/affiliates_ddl.png)](https://districtdatalabs.com/)
```
Comment (Kautumn06) [inline README.md]: Here, I updated the links to the new logo images @bbengfort resized.

File: README.md [hunk 2]
```diff
@@ -118,6 +118,6 @@ We would be glad if you used Yellowbrick in your scientific publications! If you
...
  
-[![NumFOCUS Affiliated Project](docs/images/readme/numfocus.png)](https://numfocus.org)
+[![NumFOCUS Affiliated Project](docs/images/readme/affiliates_numfocus.png)](https://numfocus.org)
```
Comment (Kautumn06) [inline README.md]: Updated the link to the updated NumFOCUS logo.

File: docs/api/cluster/elbow.rst [hunk 3]
```diff
@@ -53,7 +53,7 @@ The ``KElbowVisualizer`` also displays the amount of time to train the clusterin
     visualizer.fit(X)        # Fit the data to the visualizer
     visualizer.poof()        # Draw/show/poof the data
 
-By default, the parameter ``locate_elbow`` is set to `True`, which automatically find the "elbow" which likely corresponds to the optimal value of k using the "knee point detection algorithm". However, users can turn off the feature by setting ``locate_elbow=False``.You can read about the implementation of this algorithm at `Knee point detection in Python <https://github.com/arvkevi/kneed>`_ by Kevin Arvai. 
+By default, the parameter ``locate_elbow`` is set to ``True``, which automatically find the "elbow" which likely corresponds to the optimal value of k using the "knee point detection algorithm". However, users can turn off the feature by setting ``locate_elbow=False``. You can read about the implementation of this algorithm at "`Knee point detection in Python <https://github.com/arvkevi/kneed>`_" by Kevin Arvai.
```
Comment (Kautumn06) [inline docs/api/cluster/elbow.rst]: Added additional backticks to ensure the text rendered properly.

File: yellowbrick/cluster/elbow.py [hunk 4]
```diff
@@ -71,7 +71,7 @@ def distortion_score(X, labels, metric="euclidean"):
         The metric to use when calculating distance between instances in a
         feature array. If metric is a string, it must be one of the options
         allowed by `sklearn.metrics.pairwise.pairwise_distances
-        <https://bit.ly/2MSnF6J>`_
+        <http://bit.ly/2Z7Dxnn>`_
```
Comment (Kautumn06) [inline yellowbrick/cluster/elbow.py]: Created a new bitly link since the original was broken.

File: yellowbrick/cluster/elbow.py [hunk 5]
```diff
@@ -135,12 +135,12 @@ class KElbowVisualizer(ClusteringScoreVisualizer):
     average score for all clusters. By default, the ``distortion`` score is
     computed, the sum of square distances from each point to its assigned
     center. Other metrics can also be used such as the ``silhouette`` score,
-    the mean silhouette  coefficient for all samples or the
```
Comment (Kautumn06) [inline yellowbrick/cluster/elbow.py]: Deleted extra space

File: yellowbrick/cluster/elbow.py [hunk 6]
```diff
@@ -220,14 +220,14 @@ class KElbowVisualizer(ClusteringScoreVisualizer):
...
-    in a Haystack <https://raghavan.usc.edu//papers/kneedle-simplex11.pdf>`_.
+    `Robert Gove's Block website <https://bl.ocks.org/rpgove/0060ff3b656618e9136b>`_.
+    For more on the knee point detection algorithm see the paper `"Finding a "kneedle"
+    in a Haystack" <https://raghavan.usc.edu//papers/kneedle-simplex11.pdf>`_.
 
     .. seealso:: The scikit-learn documentation for the `silhouette_score
         <https://bit.ly/2LYWjYb>`_ and `calinski_harabasz_score
-        <https://bit.ly/2ItAgts>`_. The default, `distortion_score`, is
-        implemented in`yellowbrick.cluster.elbow`.
+        <https://bit.ly/2ItAgts>`_. The default, ``distortion_score``, is
```
Comment (Kautumn06) [inline yellowbrick/cluster/elbow.py]: Added additional backticks

File: yellowbrick/cluster/icdm.py [hunk 7]
```diff
@@ -150,7 +150,7 @@ class InterclusterDistance(ClusteringScoreVisualizer):
 
     Notes
     -----
-    Currently the only two embeddings supportted are MDS and TSNE. Soon to
```
Comment (Kautumn06) [inline yellowbrick/cluster/icdm.py]: Fixed typo

File: yellowbrick/text/dispersion.py [hunk 8]
```diff
@@ -285,7 +285,7 @@ def dispersion(
 ):
     """ Displays lexical dispersion plot for words in a corpus
 
-    This helper function is a quick wrapper to utilize the DisperstionPlot
```
Comment (Kautumn06) [inline yellowbrick/text/dispersion.py]: Fixed typo

File: yellowbrick/cluster/elbow.py [hunk 9]
```diff
@@ -135,12 +135,12 @@ class KElbowVisualizer(ClusteringScoreVisualizer):
...
     computed, the sum of square distances from each point to its assigned
     center. Other metrics can also be used such as the ``silhouette`` score,
-    the mean silhouette  coefficient for all samples or the
+    the mean silhouette coefficient for all samples or the
     ``calinski_harabasz`` score, which computes the ratio of dispersion between
     and within clusters.
 
     When these overall metrics for each model are plotted, it is possible to
-    visually determine the best value for K. If the line chart looks like an
+    visually determine the best value for k. If the line chart looks like an
```
Comment (Kautumn06) [inline yellowbrick/cluster/elbow.py]: I changed the `K` value to lowercase to match the rest of the docstring.

File: setup.py [hunk 10]
```diff
@@ -67,7 +67,7 @@
 PROJECT      = os.path.abspath(os.path.dirname(__file__))
 REQUIRE_PATH = "requirements.txt"
 VERSION_PATH = os.path.join(PACKAGE, "version.py")
-PKG_DESCRIBE = "DESCRIPTION.rst"
+PKG_DESCRIBE = "DESCRIPTION.md"
```
Comment (Kautumn06) [inline setup.py]: Updated `setup.py` to reference the new `DESCRIPTION.md` file.

**Comment by rebeccabilbro** [comment]
Looks like some great progress on this @Kautumn06! Just let me know when you're ready for a review!

**Pull request merged**